components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/util/CaptchaUtil.java

```diff
@@ -258,8 +260,22 @@ public static boolean isValidCaptcha(String reCaptchaResponse) throws CaptchaExc
         try {
             try (InputStream in = entity.getContent()) {
                 JsonObject verificationResponse = new JsonParser().parse(IOUtils.toString(in)).getAsJsonObject();
-                if (verificationResponse == null || verificationResponse.get("success") == null ||
-                        !verificationResponse.get("success").getAsBoolean()) {
+                final double score = verificationResponse.get(CaptchaConstants.CAPTCHA_SCORE).isJsonNull() ? -1
+                        : verificationResponse.get(CaptchaConstants.CAPTCHA_SCORE).getAsDouble();
+                final boolean success = !verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS).isJsonNull() &&
+                        verificationResponse.get(CaptchaConstants.CAPTCHA_SUCCESS).getAsBoolean();
+                if (score >= 0) {
+                    // reCAPTCHA v3 response contains score
+                    if (log.isDebugEnabled()) {
+                        log.debug("reCAPTCH v3 success:" + success + ", action:" +
+                                verificationResponse.get("action") + ", score:" + score);
+                    }
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("reCAPTCHA v2 success:" + success + ", action:" + verificationResponse.get("action"));
+                    }
+                }
+                if (!success ||  score < scoreThreshold) {
```

Comment:
we are assigning -1 for v2. So it will be always less than the threshold

Reply:
shall I use 2 or something?

Reply:
shall we assign the value only if it is coming in the response? and for v2 it will be null always